tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -1180,34 +1177,26 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
             }
           }
 
-        case Try(block, errVar, handler, finalizer) =>
+        /* The Google Closure Compiler used to wrongly eliminate finally blocks,
+         * if the catch block throws an exception.
+         * Issues: #563, google/closure-compiler#186
+         *
+         * Therefore, we do not special case TryFinally(TryCatch(...)).
+         * We should do this now, since GCC seems to have fixed it (#2619).
+         */
+
+        case TryCatch(block, errVar, handler) =>
           extractLet { newLhs =>
             val newBlock = pushLhsInto(newLhs, block, tailPosLabels)
-            val newHandler =
-              if (handler == EmptyTree) js.EmptyTree
-              else pushLhsInto(newLhs, handler, tailPosLabels)
-            val newFinalizer =
-              if (finalizer == EmptyTree) js.EmptyTree
-              else transformStat(finalizer, tailPosLabels)
-
-            if (newHandler != js.EmptyTree && newFinalizer != js.EmptyTree) {
-              /* The Google Closure Compiler wrongly eliminates finally blocks, if
-               * the catch block throws an exception.
-               * Issues: #563, google/closure-compiler#186
-               *
-               * Therefore, we desugar
-               *
-               *   try { ... } catch { ... } finally { ... }
-               *
-               * into
-               *
-               *   try { try { ... } catch { ... } } finally { ... }
-               */
-              js.Try(js.Try(newBlock, errVar, newHandler, js.EmptyTree),
-                  errVar, js.EmptyTree, newFinalizer)
-            } else {
-              js.Try(newBlock, errVar, newHandler, newFinalizer)
-            }
+            val newHandler = pushLhsInto(newLhs, handler, tailPosLabels)
+            js.Try(newBlock, errVar, newHandler, js.EmptyTree)
+          }
+
+        case TryFinally(block, finalizer) =>
+          extractLet { newLhs =>
+            val newBlock = pushLhsInto(newLhs, block, tailPosLabels)
+            val newFinalizer = transformStat(finalizer, tailPosLabels)
+            js.Try(newBlock, js.Ident("dummy"), js.EmptyTree, newFinalizer)
```

Comment:
Should we file a follow up issue to try and get rid of `javascript.Trees.EmptyTree` as well?